src/java.base/share/classes/jdk/internal/module/ModuleReferences.java

```diff
@@ -236,6 +244,27 @@ static JarFile newJarFile(Path path) {
         JarModuleReader(Path path, URI uri) {
             this.jf = newJarFile(path);
             this.uri = uri;
+            this.resource = new JDKResource() {
```

Comment:
Would it be cleaner to implement this JDKResource subclass as a named class in `JarFileCRaCSupport` and wrap this logic so we have something like:
```
this.resource = JarFileCRaCSupport.wrapAndRegisterJarFile(this.jf);
```

It centralizes more of the Jar file support in one place....

Reply:
Indeed, I've created ClassLoaderJarFile for that purpose. Moving the whole class to be used by ModuleReferences.